waiter/src/waiter/state/responder.clj

```diff
@@ -226,8 +226,12 @@
       (log/info "cleaning up work-stealing offers before exiting")
       ; cleanup state by rejecting any outstanding work-stealing offers, return nil since :exit was sent
       (doseq [work-stealing-offer (vec work-stealing-queue)]
-        (complete-work-stealing-offer! service-id work-stealing-offer :rejected work-stealing-received-in-flight-counter)))
-    current-state))
+        (complete-work-stealing-offer! service-id work-stealing-offer :rejected work-stealing-received-in-flight-counter))
+      ;; return nil to trigger responder termination
+      nil)
```

Comment:
Is this necessary? doseq returns nil https://clojuredocs.org/clojure.core/doseq

Reply:
Not necessary, I wanted the `nil` return to be explicit and not be affected if we decide to change the `doseq` to something else later.
It is similar in spirit to when we return `nil` explicitly after a `log` statement even though the log function/macro returns `nil`.